static/js/components/EmploymentForm.js

```diff
@@ -120,9 +120,9 @@ class EmploymentForm extends ProfileFormFields {
     if ( ui.workHistoryEdit === true ) {
       let workHistoryRows = [];
       if ( !_.isUndefined(work_history) ) {
-        workHistoryRows = Object.entries(work_history).filter(([,entry]: [string, WorkHistoryEntry]) =>
-          entry.id !== undefined
-        ).map(([i, entry]) => this.jobRow(entry, i));
+        workHistoryRows = work_history.map((entry, index) => (
```

Comment:
Hmm, flow doesn't seem to detect any problem if I change `index` from `number` to `string` for `jobRow()`. Anyway you're right that it should be able to infer this